internal/pkg/action/operator_install.go

```diff
@@ -7,11 +7,11 @@ import (
 	"strings"
 	"time"
 
+	sets "github.com/noahschumacher/go-collections/collections"
 	v1 "github.com/operator-framework/api/pkg/operators/v1"
 	"github.com/operator-framework/api/pkg/operators/v1alpha1"
 	operatorsv1 "github.com/operator-framework/operator-lifecycle-manager/pkg/package-server/apis/operators/v1"
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/sets"
```

Comment:
Curious why we're switching from this package and introducing a new dependency?

Reply:
The go linter didn't like `sets.String`.

Reply:
> The go linter didn't like sets.String.

Something reasonable it didn't like? I haven't seen any issues like that before. Did we add some new linting we really want or did something in 0.26.1 change to trip the linter?

Reply:
The new sets package uses generics; rather than write our own.

After updating go.mod/go.sum, I was getting (I had to revert these changes to demonstrate):
```
tshort@cube:~/git/operator-framework/kubectl-operator (OLM-2869 +)$ make lint
source ./scripts/fetch.sh; fetch golangci-lint 1.50.1 && ./bin/golangci-lint --timeout 3m run
WARN [runner] The linter 'golint' is deprecated (since v1.41.0) due to: The repository of the linter has been archived by the owner.  Replaced by revive.
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref.
WARN [runner] The linter 'interfacer' is deprecated (since v1.38.0) due to: The repository of the linter has been archived by the owner.
internal/pkg/action/operator_install.go:84:74: SA1019: sets.String is deprecated: use generic Set instead. new ways: s1 := Set[string]{} s2 := New[string]() (staticcheck)
func (i *OperatorInstall) possibleInstallModes(watchNamespaces []string) sets.String {
                                                                         ^
internal/pkg/action/operator_install.go:157:99: SA1019: sets.String is deprecated: use generic Set instead. new ways: s1 := Set[string]{} s2 := New[string]() (staticcheck)
func (i OperatorInstall) validateOperatorGroup(og v1.OperatorGroup, operatorInstallModes, desired sets.String) error {
                                                                                                  ^
internal/pkg/action/operator_install.go:200:57: SA1019: sets.String is deprecated: use generic Set instead. new ways: s1 := Set[string]{} s2 := New[string]() (staticcheck)
func (i *OperatorInstall) getTargetNamespaces(supported sets.String) []string {
                                                        ^
make: *** [Makefile:51: lint] Error 1
tshort@cube:~/git/operator-framework/kubectl-operator (OLM-2869 +)$

Reply:
It could be due to the updated `k8s.io/apimachinery` package being updated from v0.25.4 to v0.26.1

Reply:
Yeah, that's it. But it looks like apimachinery has a generic sets.Set type we should switch to.

https://github.com/kubernetes/apimachinery/blob/b4dea92b265131dae13c1e9ebb0a31821bf5daf8/pkg/util/sets/set.go#L24

Reply:
Maybe we should handle this in a separate PR though?